### ISSUE TITLE
Performance issues while playing dota 2

### ISSUE THREAD

**Issue by star9999999** [ValveSoftware/Dota2-Gameplay#7028]
### Description

fps droping like crazy and game feels stuttering ! 
Is it Cpu related or network related or the game related ?
Any solution ?
PC spec 5700x + 3060 + 32 gb ram non oc

### Screenshots

![Screenshot (203)](https://user-images.githubusercontent.com/123701783/220439127-cca553ad-b413-49d9-94a6-8512ed1ee36b.png)

**Comment by Fix-it3** [comment]
Yes,I'm still having this issue woth performance.I get disconnected from the match just before loading and doesn't reconnect,many people in SD are having this issue that's why they can't get out.It's a server problem for sure.#Valve Really needs to work on their game! they are losing player and will keep losong if they don't pay attention to their game!

**Comment by Omnomnommmm** [comment]
I'm also facing the same steamnetworkingsockets lock issue, it's so annoying. Key presses don't register sometimes when it happens.

**Comment by LinGG** [comment]
if you are using a laptop it might be a throttle issue, but I got absolutely new PC, it started to occur like a month ago, will try to intsall another windows to check, tried everything.

**Comment by star9999999** [comment]
> if you are using a laptop it might be a throttle issue, but I got absolutely new PC, it started to occur like a month ago, will try to intsall another windows to check, tried everything.

I am using pc ! Also found it it is the game which is poorly optimized ! I am not having any issues in other games may it be single player moba or fps Just Dota ! also there are many issues with the game . It is not our os or pc or laptop the game is the culprit.

**Comment by Omnomnommmm** [comment]
Same, using PC too. Wish there was a way to figure out exactly what is going on.

**Comment by LinGG** [comment]
ok guys, I just easily fixed it. Initially my problem was NOT only in dota, yesterday I tried to play Elden Ring, and I got so many stutters so I decided to check out what's going on, and apparently it is all about motherboard did not run my new 12600k processor as it should, so I updated my MB BIOS to the latest version, which is date this month (march) and I got 0 stutters both in Elden Ring and Dota 2. So seems like it is not about the dota, but 99% it is all about new processors architecutre (probably E-Cores used instead of P-cores)

My specs: 
Win 10, Intel 12600k, MSI Z690 Tomahawk, MSI RTX 3060 Ti

**Comment by Omnomnommmm** [comment]
@LinGG I just updated my bios last night. Will try tonight and update. Another thing I tried last night before updating the bios that seems to have fixed the steamnetworkingsockets holding issue was that I added a few launch options. I am not 100% sure which one exactly helped. I added:

1) -novid 
2) -novr 
3) -dx9 
4) +map_enable_background_maps 0

Tonight I will test by removing all 4 to see if bios updating itself does the trick.

**Comment by star9999999** [comment]
> @LinGG I just updated my bios last night. Will try tonight and update. Another thing I tried last night before updating the bios that seems to have fixed the steamnetworkingsockets holding issue was that I added a few launch options. I am not 100% sure which one exactly helped. I added:
> 
> 1. -novid
> 2. -novr
> 3. -dx9
> 4. +map_enable_background_maps 0
> 
> Tonight I will test by removing all 4 to see if bios updating itself does the trick.
> 
> My specs: Win 11, Ryzen 5 2600, ASUS PRIME A320M-K (M), GTX 1060 Zotac

it is not ur pc ! the game is not that optimized wheather u update bios or not there are going to be issue if they do not clean the game and make it optimized.

**Comment by LinGG** [comment]
> > @LinGG I just updated my bios last night. Will try tonight and update. Another thing I tried last night before updating the bios that seems to have fixed the steamnetworkingsockets holding issue was that I added a few launch options. I am not 100% sure which one exactly helped. I added:
> > 
> > 1. -novid
> > 2. -novr
> > 3. -dx9
> > 4. +map_enable_background_maps 0
> > 
> > Tonight I will test by removing all 4 to see if bios updating itself does the trick.
> > My specs: Win 11, Ryzen 5 2600, ASUS PRIME A320M-K (M), GTX 1060 Zotac
> 
> it is not ur pc ! the game is not that optimized wheather u update bios or not there are going to be issue if they do not clean the game and make it optimized.

I do NOT agree with it, these errors are 100% indication that something is off with your PC, and in my case it was obviously bad BIOS firmware, I just updated it and everything is smooth as it should be.

**Comment by star9999999** [comment]
nothing is wrong with my pc ! My BIOS is also latest and not only I have issues but multiple people on reddit too.
And from reddit I got that I should post error and bugs on github.
I play multiple games and some of the example are Lol , Valorant , CSGO etc 
No issues what so ever with thread starvation also there are 2 error in the same picture which I solved thanks to reddit which was becasue of the game.
I hope u can find it ^^

**Comment by Omnomnommmm** [comment]
Yo i removed the launch options and i haven't gotten the steamnetworksockets issue. Updating the bios seems to have done the trick

**Comment by eyup-taskin** [comment]
Same Issue on my laptop with i9-13900HX and rtx4080 Laptop GPU. Happening every 20 secs and every game dota is unplayable now
